### PR TITLE
Respect editorconfig var preference in CA1854

### PR DESCRIPTION
**Pull request by Youssef1313**
Part of #6022

### CODE REVIEW

File: src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpPreferDictionaryTryGetValueFixer.cs [hunk 1]
```diff
@@ -35,6 +36,13 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
                 return;
             }
 
+            var model = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            if (model.GetTypeInfo(dictionaryAccess).Type is not { } type)
+            {
+                return;
```
Comment (buyaa-n) [inline src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpPreferDictionaryTryGetValueFixer.cs:42]: Not sure if the type is always available, probably should use `var` if it is not

**Comment by buyaa-n** [comment]
Thanks @Youssef1313, looks good to me except it is not covered the variable name duplication part of https://github.com/dotnet/roslyn-analyzers/issues/6022, could you cover that with this PR? We can still merge this PR without that and leave the issue open for that part

**Comment by Youssef1313** [comment]
I'll take a look at the duplication issue later. The fix will likely be using `LookupSymbols` iteratively (`value`, value1`, `value2`, `value3`, ...) until we find a good name.

**Review by buyaa-n — APPROVED** [review]
LGTM, thanks @Youssef1313!

**Pull request merged**